invokeai/frontend/web/src/features/controlLayers/store/types.ts

```diff
@@ -368,6 +368,32 @@ const zCanvasRasterLayerState = zCanvasEntityBase.extend({
   position: zCoordinate,
   opacity: zOpacity,
   objects: z.array(zCanvasObjectState),
+  // Optional per-layer color adjustments (simple + curves). When null/undefined, no adjustments are applied.
+  adjustments: z
```

Comment:
Suggest modularizing this schema a bit eg:
- `zSimpleAdjustmentConfig`
- `zCurvesAdjustmentConfig`
- `zCanvasRasterLayerAdjustments`

```ts
const zChannelName = z.enum(['master', 'r', 'g', 'b']);
const zChannelConfig = z.array(z.tuple([z.number().int().min(0).max(255), z.number().int().min(0).max(255)])).min(2);
const zCurvesConfig = z.record(zChannelName, zChannelConfig);
type ChannelName = z.infer<typeof zChannelName>
type ChannelConfig = z.infer<typeof zChannelConfig>
type CurvesConfig = z.infer<typeof zCurvesConfig>
```

Then compose the schemas. Also makes it easier to get the type for just the simple/curves adjust config for use in the reducers

Also, if curves and simple are intended to be mutually exclusive, they should probably be two separate schemas.